lib/core/test/unit/Cardano/Wallet/DB/Arbitrary.hs

```diff
@@ -319,161 +329,189 @@ instance Arbitrary SlotId where
+                                 Address
+-------------------------------------------------------------------------------}
+
+instance Arbitrary Address where
+    arbitrary = oneof
+        [ pure $ Address "ADDR01"
+        , pure $ Address "ADDR02"
+        , pure $ Address "ADDR03"
+        , pure $ Address "ADDR04"
+        , pure $ Address "ADDR05"
+        , pure $ Address "ADDR06"
+        , pure $ Address "ADDR07"
+        , pure $ Address "ADDR08"
+        , pure $ Address "ADDR09"
+        , pure $ Address "ADDR10"
+        ]
+
+instance Arbitrary (Index 'Soft 'AddressK) where
     shrink _ = []
-    arbitrary = elements rootKeysSeq
+    arbitrary = arbitraryBoundedEnum
 
-instance Arbitrary (RndKey 'RootK XPrv) where
+instance Arbitrary (Index 'Hardened 'AccountK) where
     shrink _ = []
-    arbitrary = elements rootKeysRnd
+    arbitrary = arbitraryBoundedEnum
 
-instance Arbitrary (Hash "encryption") where
+instance Arbitrary (Index 'Hardened 'AddressK) where
     shrink _ = []
+    arbitrary = arbitraryBoundedEnum
+
+{-------------------------------------------------------------------------------
+                              Sequential State
+-------------------------------------------------------------------------------}
+
+instance Arbitrary (SeqState DummyTarget) where
+    shrink (SeqState intPool extPool ixs) =
+        (\(i, e, x) -> SeqState i e x) <$> shrink (intPool, extPool, ixs)
     arbitrary = do
-        InfiniteList bytes _ <- arbitrary
-        return $ Hash $ BS.pack $ take 32 bytes
+        SeqState <$> arbitrary <*> arbitrary <*> arbitrary
 
-genRootKeysSeq :: Gen (SeqKey 'RootK XPrv)
-genRootKeysSeq = do
-    (s, g, e) <- (,,)
-        <$> genPassphrase @"seed" (16, 32)
-        <*> genPassphrase @"generation" (0, 16)
-        <*> genPassphrase @"encryption" (0, 16)
-    return $ Seq.generateKeyFromSeed (s, g) e
+instance Arbitrary (SeqKey 'RootK XPrv) where
+    shrink _ = []
+    arbitrary = elements rootKeysSeq
 
-genRootKeysRnd :: Gen (RndKey 'RootK XPrv)
-genRootKeysRnd = Rnd.generateKeyFromSeed
-    <$> genPassphrase @"seed" (16, 32)
-    <*> genPassphrase @"encryption" (0, 16)
+-- FIXME:
+-- Store pending change outside of the sequential state. Pending state is not
+-- affected by rollbacks and therefore, managing with checkpoints makes testing
+-- fairly difficult as many nice properties fail if we consider non empty change
+-- indexes. For example:
+--
+-- - put checkpoint with pending indexes
+-- - put checkpoint with empty pending indexes
+-- - read latest checkpoint
+--
+-- We would expect the latest checkpoint to be the one we just inserted, but
+-- currently, it'll also contains the pending indexes from the previously
+-- inserted checkpoints!
+--
+--    shrink =
+--        map Seq.pendingIxsFromList . shrink . Seq.pendingIxsToList
+--    arbitrary =
+--        Seq.pendingIxsFromList . Set.toList <$> arbitrary
+instance Arbitrary (Seq.PendingIxs) where
+    arbitrary = pure Seq.emptyPendingIxs
```

Comment:
This is the only change in logic in this file (rest is just functions declarations being moved around). 
We need to review how pending indexes are stored before we can enable arbitrary pending indexes in tests. Will do in another PR.